File: app/src/main/java/com/geovanni/banks/bussiness/entityDao/roomDataBase/ManagerDbAsync.java
package com.geovanni.banks.bussiness.entityDao.roomDataBase;

import android.os.AsyncTask;

import com.geovanni.banks.bussiness.entityDao.interfaces.IServiceBanksDao;
import com.geovanni.banks.bussiness.models.ServiceBanksResponse;

import java.util.List;

public class ManagerDbAsync {

    public static class InsertDataToDbAsync extends AsyncTask<Void, Void, Void> {

        private final IServiceBanksDao banksDao;
        private final List<ServiceBanksResponse> serviceBanksResponses;

        public InsertDataToDbAsync(BankRoomDataBase db, List<ServiceBanksResponse> serviceBanksResponses) {
            this.banksDao = db.banksDao();
            this.serviceBanksResponses = serviceBanksResponses;
        }

        @Override
        protected Void doInBackground(Void... voids) {
            for (ServiceBanksResponse bank : serviceBanksResponses) {
                banksDao.insert(bank);
            }
            return null;
        }
    }

    public static class GetAllDataFromDbAsync extends AsyncTask<Void, Void, List<ServiceBanksResponse>> {
        private final IServiceBanksDao banksDao;

        public GetAllDataFromDbAsync(BankRoomDataBase db) {
            banksDao = db.banksDao();
        }

        @Override
        protected List<ServiceBanksResponse> doInBackground(Void... voids) {
            return banksDao.getAll();
        }
    }
}

File: app/src/main/java/com/geovanni/banks/bussiness/entityDao/roomDataBase/ManagerDbAsync.kt
package com.geovanni.banks.bussiness.entityDao.roomDataBase

import android.os.AsyncTask
import com.geovanni.banks.bussiness.entityDao.interfaces.IServiceBanksDao
import com.geovanni.banks.bussiness.models.ServiceBanksResponse

class ManagerDbAsync {
    class InsertDataToDbAsync(db: BankRoomDataBase, serviceBanksResponses: List<ServiceBanksResponse>) : AsyncTask<Void?, Void?, Void?>() {

        private val banksDao: IServiceBanksDao = db.banksDao()
        private val serviceBanksResponses: List<ServiceBanksResponse> = serviceBanksResponses

        override fun doInBackground(vararg params: Void?): Void? {
            for (bank in serviceBanksResponses) {
                banksDao.insert(bank)
            }
            return null
        }

    }

    class GetAllDataFromDbAsync(db: BankRoomDataBase) : AsyncTask<Void?, Void?, List<ServiceBanksResponse?>?>() {

        private val banksDao: IServiceBanksDao = db.banksDao()

        override fun doInBackground(vararg params: Void?): List<ServiceBanksResponse?>? {
            return banksDao.all
        }

    }
}